### DISCUSSION TITLE
Add label to onboarding pull requests

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#24871] [Suggest an Idea]
### What would you like Renovate to be able to do?

A user asked:

> The `RENOVATE_ADD_LABELS` env var is great to add labels to the MR's, however is there a way of adding specific ones to the onboarding MR's, so I can add a `'onboarding'` label to only those?

@rarkins liked this idea. I'm creating this issue to discuss the requirements some more, and to track this idea. 😄 

### If you have any ideas on how this should be implemented, please tell us here.

We already have a `addLabels` feature, maybe the code for that can inspire the new code?

- [ ] Write code for new feature
- [ ] Create Renovate built-in preset to add a `onboarding` label that users can `extend` from
- [ ] Mention onboarding labels in the [`addLabels` docs](https://docs.renovatebot.com/configuration-options/#addlabels)
- [ ] Explain in the docs how a maintainer can tell Renovate to add the `onboarding` label to new onboarding PRs

### Is this a feature you are interested in implementing yourself?

No

**Comment by HonkingGoose** [comment]
Adding labels to a onboarding PR that you'll usually get _once per repository_, is probably low-priority, or even `wont-do` territory. What do you think @rarkins?

**Comment by slonim-greg** [comment]
I wanted to inquire if this is something that you can consider implementing or reviewing a contribution for.

**Context**
In our organisation, there are strict pre-flight rules on MRs/PRs, specifically, their correlation with Jira tickets, among other policies. Adding a label/labels to these onboarding MRs will allow us to remove friction in onboarding repos for the bot (where we can bypass these policies automatically based on labels). Please share your thougths.

**Reply by HonkingGoose** [reply]
I will let a maintainer repond _officially_ for the project. 😉 

In the meantime, if you're self-hosting you may be able to use `onboardingConfig` to add a label to Renovate onboarding PRs:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "labels": ["bypass-policy-rules"],
}
```

### References

- [Renovate docs, self-hosting config `onboardingConfig`](https://docs.renovatebot.com/self-hosted-configuration/#onboardingconfig)
- [Renovate docs, `labels` config option](https://docs.renovatebot.com/configuration-options/#labels)

**Reply by slonim-greg** [reply]
This worked brilliantly and resolved our problems. Thank you!

**Reply by HonkingGoose** [reply]
We now support adding a label to onboarding PRs if you _self-host_ Renovate, see the comment chain above.

`@rarkins` do you want to create a similar feature for the _Mend-hosted Renovate app_ config?

**Reply by rarkins** [reply]
This might possibly work already using `inheritConfig`

**Reply by HonkingGoose** [reply]
`inheritConfig` is a _self-hosted_ config option. That does not work for those on a Mend-hosted app.

I skimmed our repository config options, and searched for `inherit` and `onboarding`. I did not find a repository config option to set a label for onboarding PRs.

I'll let you decide what to do based on this info. 😉

**Reply by rarkins** [reply]
We are adding inheritConfig support to the hosted app too, so long as users use the default file naming convention

**Reply by HonkingGoose** [reply]
Two questions for `@rarkins`:

- Do we have an issue on the `renovatebot/renovate` repository to _directly_ track "add `inheritConfig` support to the hosted app"?
- Or is improvement already tracked in a Mend-internal todo?

## Open issues I found that seem relevant

Searching for `inheritConfig` gives _no_ open issues. But searching `inherit config` leads to:

- https://github.com/renovatebot/renovate/issues/31278

I skimmed the open issues labeled with `core:config`. I think these are somewhat relevant:

- https://github.com/renovatebot/renovate/issues/29536
- https://github.com/renovatebot/renovate/issues/21110
- https://github.com/renovatebot/renovate/issues/11791
- https://github.com/renovatebot/renovate/issues/10682

**Reply by rarkins** [reply]
I think maybe we can have open discussions when it relates to the Mend app, and reserve issues for things which are related to code in this repo.

@justo-mend what's the status of inheritConfig support in the Mend app?

**Reply by justo-mend** [reply]
The inheritConfig feature is enabled and working on the cloud-hosted Mend Apps. (Documentation is [pending PR](https://github.com/justo-mend/renovate/pull/3/files) approval)

To use inherited config:
1. Include the config in a file called "org-inherited-config.json" in a repo called "renovate-config" within the same org.
2. Ensure the "renovate-config" org has Renovate enabled

Note: the feature only takes effect when the org-inherited-config.json file is added or modified AFTER the repo (renovate-config) has had Renovate installed. (Onboarding not required)

**Reply by HonkingGoose** [reply]
The documentation changes are published now:

- https://github.com/renovatebot/renovate/pull/31479
